tests/Agent/IntegrationTests/Shared/ReflectionHelpers/DynamicMethodExecutor.cs

```diff
@@ -31,7 +32,13 @@ public void ExecuteDynamicMethod(MethodInfo method, string[] args)
                 }
             }
 
-            method.Invoke(objInst, paramValues.ToArray());
+            var returnValue = method.Invoke(objInst, paramValues.ToArray());
```

Comment:
This makes sure the caller wait for `Task` based method invocation to finish before moving on execution other methods.